src/mean.jl

```diff
@@ -29,99 +9,139 @@ julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
         p.mean = p.counts ./ sum(p.counts)
         # variance of proportion
         p.var = design.fpc .* p.mean .* (1 .- p.mean) ./ (design.sampsize - 1)
-        p.sem = sqrt.(p.var)
+        p.se = sqrt.(p.var)
         return select(p, Not([:counts, :var]))
     end
-    return DataFrame(mean=mean(design.data[!, x]), sem=sem(x, design))
+    return DataFrame(mean=mean(design.data[!, x]), SE=se(x, design))
 end
 
+"""
+    mean(x, design)
+Estimate the population mean of a variable of a simple random sample, and the corresponding standard error.
+
+```jldoctest
+julia> apisrs = load_data("apisrs");
+
+julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
+
+julia> mean([:api00, :api99], srs)
+2×3 DataFrame
+ Row │ names   mean     SE     
+     │ String  Float64  Float64 
+─────┼──────────────────────────
+   1 │ api00   656.585  9.24972
+   2 │ api99   624.685  9.5003
+```
+"""
 function mean(x::Vector{Symbol}, design::SimpleRandomSample)
-    means_list = []
-    for i in x
-        push!(means_list, mean(i, design))
-    end
-    df = reduce(vcat, means_list)
+    df = reduce(vcat, [mean(i, design) for i in x])
     insertcols!(df, 1, :names => String.(x))
     return df
 end
 
 """
-Inner method for `by`
-"""
-# Inner methods for `by`
-function sem_by(x::AbstractVector, design::SimpleRandomSample)
-    # domain size
-    dsize = length(x)
-    # sample size
-    ssize = design.sampsize
-    # fpc
-    fpc = design.fpc
-    # variance of the mean
-    variance = (dsize / ssize)^(-2) / ssize * fpc * ((dsize - 1) / (ssize - 1)) * var(x)
-    # return the standard error
-    return sqrt(variance)
-end
+Estimate domain-wise mean. 
+```jldoctest
+julia> using Survey; 
 
+julia> srs = load_data("apisrs"); 
+
+julia> srs = SimpleRandomSample(srs; popsize = :fpc);
+
+julia> mean(:api00, :cname, srs) |> first
+DataFrameRow
+ Row │ cname     mean     SE     
+     │ String15  Float64  Float64 
+─────┼────────────────────────────
+   1 │ Kern        573.6  42.8026
+```
 """
-Inner method for `by` for SimpleRandomSample
-"""
-function mean(x::AbstractVector, design::SimpleRandomSample, weights)
-    return DataFrame(mean=Statistics.mean(x), sem=sem_by(x, design))
+function mean(x::Symbol, by::Symbol, design::SimpleRandomSample) 
+    function domain_mean(x::AbstractVector, design::SimpleRandomSample, weights)
+        function se(x::AbstractVector, design::SimpleRandomSample)
+            nd = length(x)  # domain size
+            n = design.sampsize
+            fpc = design.fpc
+            variance = (nd / n)^(-2) / n * fpc * ((nd - 1) / (n - 1)) * var(x)
+            return sqrt(variance)
+        end
+        return DataFrame(mean=Statistics.mean(x), SE=se(x, design))
+    end
+    gdf = groupby(design.data, by)
+    combine(gdf, [x, :weights] => ((a, b) -> domain_mean(a, design, b)) => AsTable)
 end
 
 """
-Inner method for `by` for StratifiedSample
 Calculates domain mean and its std error, based example 10.3.3 on pg394 Sarndal (1992)
```

Comment:
Again, I suggest removing this docstring. It's the same functionality as above, but for a different type. We can clarify that it works on `StratifiedSample` as well by adding an example in the above docstring.